ignite/cmd/model/chain_debug.go

```diff
@@ -0,0 +1,140 @@
+package cmdmodel
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	tea "github.com/charmbracelet/bubbletea"
+
+	"github.com/ignite/cli/ignite/pkg/cliui/colors"
+	cliuimodel "github.com/ignite/cli/ignite/pkg/cliui/model"
+	"github.com/ignite/cli/ignite/pkg/events"
+	"github.com/ignite/cli/ignite/pkg/xstrings"
+)
+
+const (
+	stateChainDebugStarting uint = iota
+	stateChainDebugRunning
+)
+
+var msgStopDebug = colors.Faint("Press the 'q' key to stop debug server")
+
+// NewChainDebug returns a new UI model for the chain debug command.
+func NewChainDebug(mCtx Context, bus events.Provider, cmd tea.Cmd) ChainDebug {
+	// Initialize a context and cancel function to stop execution
+	ctx, quit := context.WithCancel(mCtx.Context())
+
+	// Update the context to allow stopping by using the 'q' key
+	mCtx.SetContext(ctx)
+
+	return ChainDebug{
+		cmd:   cmd,
+		quit:  quit,
+		model: cliuimodel.NewEvents(bus),
+	}
+}
+
+// ChainDebug defines a UI model for the chain debug command.
+type ChainDebug struct {
+	cmd   tea.Cmd
+	quit  context.CancelFunc
+	state uint
+	error error
+	model cliuimodel.Events
+}
+
+func (m ChainDebug) Init() tea.Cmd {
+	return tea.Batch(m.model.WaitEvent, m.cmd)
+}
+
+func (m ChainDebug) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
```

Comment:
Let's add godoc comments for all of our exported functions

Reply:
Changeset b3b63bd